File: src/main/java/com/github/kindrat/cassandra/client/ui/fx/EnumStringConverter.java
package com.github.kindrat.cassandra.client.ui.fx;

import javafx.util.StringConverter;
import lombok.RequiredArgsConstructor;
import lombok.SneakyThrows;

@RequiredArgsConstructor
public class EnumStringConverter<T extends Enum<T>> extends StringConverter<T> {
    private final Class<T> enumClass;

    @Override
    public String toString(T object) {
        return object.toString().toUpperCase();
    }

    @Override
    @SneakyThrows
    public T fromString(String string) {
        return Enum.valueOf(enumClass, string.toUpperCase());
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/fx/EnumStringConverter.kt
package com.github.kindrat.cassandra.client.ui.fx

import javafx.util.StringConverter

class EnumStringConverter<T : Enum<T>>(private val enumClass: Class<T>) : StringConverter<T>() {

    override fun toString(value: T): String {
        return value.toString().toUpperCase()
    }

    override fun fromString(string: String): T {
        return java.lang.Enum.valueOf(enumClass, string.toUpperCase())
    }
}